app/src/main/java/org/tuerantuer/launcher/ui/screen/AllAppsScreen.kt

```diff
@@ -70,3 +142,20 @@ fun AllAppsScreenPreview() {
         )
     }
 }
+
+/**
+ * Filters the list of apps based on the search query.
+ */
+fun filter(searchQuery: String, allAppsList: List<AppItemInfo>): List<AppItemInfo> {
+    Timber.d("Carikom: Search results: ")
+    return (allAppsList.filter { appItemInfo ->
+        normalizeName(appItemInfo.name).contains(normalizeName(searchQuery))
+    })
+}
+
+/**
+ * Normalizes the name of an app by removing whitespaces and making it lowercase.
+ */
+fun normalizeName(name: String): String {
+    return name.replace("\\s+".toRegex(), "").toLowerCase(Locale.ROOT)
```

Comment:
Das `"\\s+".toRegex()` bitte aus Performancegründen irgendwo in ne globale Variable packen